src/script/lua_api/l_settings.cpp

```diff
@@ -226,6 +242,22 @@ int LuaSettings::l_set_np_group(lua_State *L)
 	return 0;
 }
 
+// set_pos(self, key, value)
+int LuaSettings::l_set_pos(lua_State *L)
+{
+    NO_MAP_LOCK_REQUIRED;
+    LuaSettings *o = checkObject<LuaSettings>(L, 1);
+
+    std::string key = std::string(luaL_checkstring(L, 2));
```

Comment:
```suggestion
    std::string key = luaL_checkstring(L, 2);
```

the copy assignment constructor should already do the job.